pkg/ipcache/ipcache.go

```diff
@@ -809,6 +811,15 @@ func (m *K8sMetadata) Equal(o *K8sMetadata) bool {
 			return false
 		}
 	}
+	if m.Workload != nil {
```

Comment:
`if m.Workload.Equals(o.Workload)` is sufficient. The `Equals` receiver already checks for `nil`.